File: 0929_Unique_Email_Addresses/0929_myAns.cpp
class Solution
{
public:
    int numUniqueEmails(vector<string> &emails)
    {
        unordered_map<string, bool> email_adress_list;
        int count = 0;

        for (string email : emails)
        {
            string forwarded_adress;
            bool is_local_name = true;
            int i = 0;
            int email_size = email.size();
            while (i < email_size)
            {
                if (is_local_name)
                {
                    if (email[i] == '.')
                    {
                        i++;
                        continue;
                    }
                    if (email[i] == '+')
                    {
                        while (email[i] != '@')
                        {
                            i++;
                        }
                    }
                    if (email[i] == '@')
                        is_local_name = false;
                }
                forwarded_adress.push_back(email[i]);
                i++;
            }

            if (email_adress_list.find(forwarded_adress) == email_adress_list.end())
            {
                count += 1;
                email_adress_list[forwarded_adress] = true;
            }
        }

        return count;
    }
};

File: 0929_Unique_Email_Addresses/0929.myans.rs
use std::collections::HashSet;

impl Solution {
    pub fn num_unique_emails(emails: Vec<String>) -> i32 {
        let mut unique_email_set = HashSet::new();
        for email in emails {
            let names:Vec<&str> = email.split('@').collect();
            let local_name:Vec<&str> = names[0].split('+').collect();
            let validated_local_name = local_name[0].replace(".", "");
            let validated_email = validated_local_name + "@" + names[1];
            unique_email_set.insert(validated_email);
        }
        unique_email_set.len() as i32
    }
}